Assignee: Martin He

### Problem 1: Kaplan-Meier estimator

In [1]:
## Load Kaplan-Meier Estimator library
library(survival)

## Import data from csv file
data <- read.csv("leuk.csv")

## Set up variables (time, condition) for 6-MP group and Placebo group (with condition of 1 indicating relapse):
time_6mp <- data$time[data$group=='6 MP']
condition_6mp <- data$cens[data$group=='6 MP']
time_placebo <- data$time[data$group=='Placebo']
condition_placebo <- data$cens[data$group=='Placebo']

1. Compute the survival function for Week 1 - 12, using the Kaplan-Meier estimator for each group.

In [2]:
## First, calculating the survival funtion for 6-MP Group:
survival_6mp <- Surv(time_6mp, condition_6mp)
fit_6mp <- survfit(survival_6mp ~ 1)
cat("Follow-up Table for 6-MP Group Children:\n")
print(summary(fit_6mp, times = 1:12))

Follow-up Table for 6-MP Group Children:
Call: survfit(formula = survival_6mp ~ 1)

 time n.risk n.event survival std.err lower 95% CI upper 95% CI
    1     21       0    1.000  0.0000        1.000        1.000
    2     21       0    1.000  0.0000        1.000        1.000
    3     21       3    0.857  0.0764        0.720        1.000
    4     17       1    0.807  0.0869        0.653        0.996
    5     16       1    0.756  0.0950        0.591        0.967
    6     13       0    0.756  0.0950        0.591        0.967
    7     12       1    0.693  0.1059        0.514        0.935
    8     11       1    0.630  0.1135        0.443        0.897
    9     10       0    0.630  0.1135        0.443        0.897
   10      9       0    0.630  0.1135        0.443        0.897
   11      7       1    0.540  0.1281        0.339        0.860
   12      6       1    0.450  0.1347        0.250        0.809


In [3]:
## Then, calculating the survival funtion for Placebo Group:
survival_placebo <- Surv(time_placebo, condition_placebo)
fit_placebo <- survfit(survival_placebo ~ 1)
cat("Follow-up Table for Placebo Group Children:\n")
print(summary(fit_placebo, times = 1:12))

Follow-up Table for Placebo Group Children:
Call: survfit(formula = survival_placebo ~ 1)

 time n.risk n.event survival std.err lower 95% CI upper 95% CI
    1     21       4   0.8095  0.0857      0.65785        0.996
    2     17       3   0.6667  0.1029      0.49268        0.902
    3     14       2   0.5714  0.1080      0.39455        0.828
    4     12       4   0.3810  0.1060      0.22085        0.657
    5      8       0   0.3810  0.1060      0.22085        0.657
    6      8       4   0.1905  0.0857      0.07887        0.460
    7      4       0   0.1905  0.0857      0.07887        0.460
    8      4       1   0.1429  0.0764      0.05011        0.407
    9      3       1   0.0952  0.0641      0.02549        0.356
   10      2       0   0.0952  0.0641      0.02549        0.356
   11      2       1   0.0476  0.0465      0.00703        0.322
   12      1       1   0.0000     NaN           NA           NA


2. What are the median survival time for the two groups, 6-MP and placebo?

In [4]:
## Calculating the median survival time for 6-MP:
median_6mp <- summary(fit_6mp)$table['median']
cat(median_6mp)

12

In [5]:
## Calculating the median survival time for Placebo:
median_placebo <- summary(fit_placebo)$table['median']
cat(median_placebo)

4

3. Perform logrank test to compare the survival distributions, using R.

In [6]:
surv_obj <- with(data, Surv(time, cens, type="right"))
logrank_test <- survdiff(surv_obj ~ group, data = data)
print(logrank_test)

Call:
survdiff(formula = surv_obj ~ group, data = data)

               N Observed Expected (O-E)^2/E (O-E)^2/V
group=6 MP    21        9       19      5.25      16.7
group=Placebo 21       21       11      9.06      16.7

 Chisq= 16.7  on 1 degrees of freedom, p= 4e-05 


4. What is the conclusion of the logrank test?

A: This study was set up a null hypothesis: survival probabilities of 6-MP and of Placebo are the same at all follow-up times.
By running the logrank test above, we check the chi-square result which is 16.7, as well as the p-value which is 0.00004.
These result means there is a significant difference in the survival probabilities between 6-MP and Placebo Group,
hence rejecting the null hypothesis.